### PR TITLE
feat: Observe silent drop event count

### PR DESCRIPTION
**Pull request by guitarrapc**
## Summary

- Expose cumulative `DroppedEventCount` on GC, ThreadPool, and timer listeners.
- Count items evicted by bounded `DropOldest` channels.
- Add boundary tests covering channel capacity and overflow behavior.
- Document the delivery and counter semantics.

## Motivation

Bounded channels intentionally drop old values when readers fall behind, but these drops were previously silent. The new counters make event loss observable without blocking producers or adding per-event allocations.

## Benchmark / Verification

- Hot-path allocation regression test remains at **0 B**
- **93 tests passed**
- Release build passed for all target frameworks with **0 warnings**
- NuGet packaging passed

### CODE REVIEW

File: src/ClrProfiler/EventListeners/GCEventListener.cs [hunk 1]
```diff
@@ -36,9 +42,11 @@ public GCEventListener(Func<GCEventStatistics, Task> onEventEmit, Action<Excepti
             SingleWriter = false,
             FullMode = BoundedChannelFullMode.DropOldest,
         };
-        _channel = Channel.CreateBounded<GCEventStatistics>(channelOption);
+        _channel = Channel.CreateBounded<GCEventStatistics>(channelOption, OnItemDropped);
     }
```
[the remark this answers is not in the document]
Comment (guitarrapc) [reply src/ClrProfiler/EventListeners/GCEventListener.cs]: This overload is available in the .NET 8, .NET 9, and .NET 10 reference assemblies (and has been available since .NET 6). I also verified the project by building each target framework individually; all three builds completed with zero warnings and zero errors. No additional System.Threading.Channels package is required.
